.github/workflows/main.yml

```diff
@@ -0,0 +1,75 @@
+name: Pipeline
+
+on:
+  pull_request:
+    branches:
+      - develop
+    types: [opened, synchronize]
+  push:
+    branches:
+      - develop
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    services:
+      postgres:
+        image: postgres:latest
+        ports:
+          - 5432:5432
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+        env:
+          POSTGRES_DB: ${{ secrets.DB_NAME }}
+          POSTGRES_USER: ${{ secrets.DB_USER }}
+          POSTGRES_PASSWORD: ${{ secrets.DB_PASSWORD }}
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.10.4'
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt
+
+    - name: Run Migrations
+      run: |
+        DB_NAME=${{ secrets.DB_NAME }} DB_USER=${{ secrets.DB_USER }} DB_PASSWORD=${{ secrets.DB_PASSWORD }} DB_HOST=${{ secrets.DB_HOST }} DB_PORT=${{ secrets.DB_PORT }} python manage.py migrate
+
+    - name: Run unit tests and notify Slack on failure
+      run: |
+        DB_NAME=${{ secrets.DB_NAME }} \
+        DB_USER=${{ secrets.DB_USER }} \
+        DB_PASSWORD=${{ secrets.DB_PASSWORD }} \
+        DB_HOST=${{ secrets.DB_HOST }} \
+        DB_PORT=${{ secrets.DB_PORT }} \
```

Comment:
Is there a better way to do this? What if there were 20, 30 or even more env variables? Would you still be writing them in a single line like this? Also, I still think there is redundancy here and can be improved.

Reply:
I handled these using env block in the workflow now, it will improve the redundancy issue and make them to write for only 1 time.